com.unity.render-pipelines.universal/Editor/Converter/Converters.cs

```diff
@@ -0,0 +1,226 @@
+        PPv2,
+        ParametricToFreeformLight,
+    }
+
+    public static class Converters
+    {
+        static Type GetContainerType(ConverterContainerId containerName)
+        {
+            switch (containerName)
+            {
+                case ConverterContainerId.BuiltInToURP:
+                    return typeof(BuiltInToURPConverterContainer);
+                case ConverterContainerId.BuiltInToURP2D:
+                    return typeof(BuiltInToURP2DConverterContainer);
+                case ConverterContainerId.UpgradeURP2DAssets:
+                    return typeof(UpgradeURP2DAssetsContainer);
+            }
+
+            return null;
+        }
+
+        static Type GetConverterType(ConverterId converterName)
+        {
+            switch (converterName)
+            {
+                case ConverterId.Material:
+                    return typeof(UniversalRenderPipelineMaterialUpgrader);
+                case ConverterId.RenderSettings:
+                    return typeof(RenderSettingsConverter);
+                case ConverterId.AnimationClip:
+                    return typeof(AnimationClipConverter);
+                case ConverterId.ReadonlyMaterial:
+                    return typeof(ReadonlyMaterialConverter);
+#if PPV2_EXISTS
+                case ConverterId.PPv2:
+                    return typeof(PPv2Converter);
+#endif
+                case ConverterId.ParametricToFreeformLight:
+                    return typeof(ParametricToFreeformLightUpgrader);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// The method that will be run when converting assets in batch mode.
+        /// </summary>
+        /// <param name="containerName">The name of the container which will be batched. All Converters in this Container will run if prerequisites are met.</param>
+        public static void RunInBatchMode(ConverterContainerId containerName)
+        {
+            Type typeName = GetContainerType(containerName);
+            if (typeName != null)
+            {
+                RunInBatchMode(typeName);
+            }
+        }
+
+        /// <summary>
+        /// The method that will be run when converting the assets in batch mode.
+        /// </summary>
+        /// <param name="containerName">The name of the container which will be batched.</param>
+        /// <param name="converterList">The list of converters that will be either included or excluded from batching. These converters need to be part of the passed in container for them to run.</param>
+        /// <param name="converterFilter">The enum that decide if the list of converters will be included or excluded when batching.</param>
+        public static void RunInBatchMode(ConverterContainerId containerName, List<ConverterId> converterList, ConverterFilter converterFilter)
+        {
+            Type containerType = GetContainerType(containerName);
+            List<Type> converterTypes = new List<Type>();
```

Comment:
Reduce reallocations by reserving enough space initially for the maximum amount of items that will end up in the List.
```suggestion
            List<Type> converterTypes = new List<Type>(converterList.Count);
```